packages/generator/src/templateHelpers.ts

```diff
@@ -243,68 +258,96 @@ export const getValue = function(name: any): string {
   return null;
 };
 
-const getProperties = function(
-  propertyShapes: model.domain.PropertyShape[],
-  filterEntriesBy: Function
+type propertyFilter = (propertyName: string) => boolean;
+
+/**
+ * Get properties of the DTO (inherited and linked) after applying the given filter criteria
+ *
+ * @param dtoTypeModel AMF model of the dto
+ * @param propertyFilter function to filter properties based on certain criteria
+ */
+const getFilteredProperties = function(
+  dtoTypeModel: model.domain.NodeShape | null | undefined,
+  propertyFilter: propertyFilter
 ): model.domain.PropertyShape[] {
-  return !propertyShapes
-    ? []
-    : propertyShapes.filter(entry => filterEntriesBy(entry));
+  const properties: model.domain.PropertyShape[] = [];
+  const existingProps: Set<string> = new Set();
+
+  while (dtoTypeModel != null) {
+    if (dtoTypeModel.properties != null && dtoTypeModel.properties.length > 0) {
+      dtoTypeModel.properties.forEach(prop => {
+        if (prop != null) {
```

Comment:
Can we move "if(prop != null)" to a filter before forEach.? 

This will avoid nested if inside forEach

Reply:
but that is similar to looping twice right? which is not efficient